builtin/stash--helper.c

```diff
@@ -634,6 +640,55 @@ static int list_stash(int argc, const char **argv, const char *prefix)
 	return ret;
 }
 
+static int show_stash(int argc, const char **argv, const char *prefix)
+{
+	int i, ret = 0;
+	struct argv_array args = ARGV_ARRAY_INIT;
+	struct argv_array args_refs = ARGV_ARRAY_INIT;
+	struct stash_info info;
+	struct option options[] = {
+		OPT_END()
+	};
+
+	argc = parse_options(argc, argv, prefix, options,
+			git_stash_helper_show_usage, PARSE_OPT_KEEP_UNKNOWN);
+
+	argv_array_push(&args, "diff");
+
+	for (i = 0; i < argc; ++i) {
+		if (argv[i][0] == '-')
+			argv_array_push(&args, argv[i]);
+		else
+			argv_array_push(&args_refs, argv[i]);
+	}
+
+	ret = get_stash_info(&info, args_refs.argc, args_refs.argv);
+	argv_array_clear(&args_refs);
+
+	if (!ret && args.argc == 1) {
```

Comment:
I am fairly certain that we should `die()` before this conditional if `ret` is non-zero, so we do not need to test for it here.

However, the way I read `test -z "$FLAGS"` in the shell script, I would assume that we have to test for `if (args.argc == 0)` here (or `if (!args.argc)`, which is kind of the convention in Git's source code).

Reply:
Because of `argv_array_push(&args, "diff");` on line 661 `args.argc` will be 1, not 0.